core/txpool/legacypool/legacypool.go

```diff
@@ -1362,8 +1376,18 @@ func (pool *LegacyPool) runReorg(done chan struct{}, reset *txpoolResetRequest,
 	}
 	pool.mu.Lock()
 	if reset != nil {
-		// Reset from the old head to the new, rescheduling any reorged transactions
-		pool.reset(reset.oldHead, reset.newHead)
+		// only reset the state root and the head of the txpool when we are running the auctioneer node.
+		// when we are not running the auctioneer node, we re-inject any re-orged transactions which is similar
+		// to the current functionality of geth
+		if pool.auctioneerEnabled {
+			// only reset from the old head to the new head.
+			// since we will be clearing the mempool, we do not need to re-inject any re-orged transactions
+			pool.resetHeadOnly(reset.oldHead, reset.newHead)
```

Comment:
why do we clear the mempool for auctioneer? wouldn't you need to add re-orged txs if the optimistic block changes to a different fork (or no, since the txs are already assumed to be sent to sequencer?)